src/exception/repo.ts

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) 2022 The Ontario Institute for Cancer Research. All rights reserved
+ *
+ * This program and the accompanying materials are made available under the terms of
+ * the GNU Affero General Public License v3.0. You should have received a copy of the
+ * GNU Affero General Public License along with this program.
+ *  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT
+ * SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+import mongoose from 'mongoose';
+import { F, MongooseUtils } from '../utils';
+import { loggerFor } from '../logger';
+import { DeepReadonly } from 'deep-freeze';
+
+const L = loggerFor(__filename);
+
+const ExceptionSchema = new mongoose.Schema({
+  schema: { type: String },
+  coreField: { type: String },
+  exceptionValue: { type: String, enum: ['Unknown', 'Missing', 'Not applicable'] },
+});
+
+const ProgramExceptionSchema = new mongoose.Schema({
+  name: { type: String, unique: true, required: true },
+  exceptions: [ExceptionSchema],
+});
+
+type ProgramExceptionDocument = mongoose.Document & ProgramException;
+
+export const ProgramExceptionModel = mongoose.model<ProgramExceptionDocument>(
+  'ProgramException',
+  ProgramExceptionSchema,
+);
+
+interface ProgramExceptionItem {
+  schema: string;
+  coreField: string;
+  exceptionValue: string;
+}
+
+export interface ProgramException {
+  name: string;
+  exceptions: ProgramExceptionItem[];
+}
+
+export interface ProgramExceptionRepository {
+  create(programException: ProgramException): Promise<DeepReadonly<ProgramException>>;
+  find(name: string): Promise<DeepReadonly<ProgramException> | undefined>;
+  update(programException: ProgramException): Promise<DeepReadonly<ProgramException> | undefined>;
+  delete(name: string): Promise<void>;
+}
+
+export const programExceptionRepository: ProgramExceptionRepository = {
+  async create(req: ProgramException) {
+    L.debug(`Creating new program exception with: ${JSON.stringify(req)}`);
+    const exception: ProgramException = { ...req };
+    const programException = new ProgramExceptionModel(exception);
```

Comment:
Why not just pass your argument directly to the model?
```suggestion
    const programException = new ProgramExceptionModel(req);
```